prov/shm/src/smr_av.c

```diff
@@ -148,8 +152,16 @@ static int smr_av_insert(struct fid_av *av_fid, const void *addr, size_t count,
 		if (ret) {
 			if (fi_addr)
 				fi_addr[i] = util_addr;
-			if (shm_id >= 0)
-				smr_map_del(&smr_av->smr_map, shm_id);
+			if (shm_id >= 0) {
+				node = ofi_rbmap_find(&smr_av->smr_map.rbmap,
+				      &smr_av->smr_map.peers[shm_id].peer.name);
+				assert(node);
+				ret = smr_map_del(&smr_av->smr_map, node);
```

Comment:
Now that both places where you call smr_map_del do a find and pass in the node, let's have the call take in the id instead and do the find in there. You can use the rbmap_find_delete call too to simplify.

Reply:
^I think this comment still applies. And I think it would resolve a potential issue with the locking here where you search the rbmap without acquiring the map lock